### PR TITLE
🌿 Introduce Fern Docs

### PR DESCRIPTION
**Pull request by dannysheridan**
This PR adds the open source, free tier of **Fern Docs** for PrivateGPT.

### CODE REVIEW

**Comment by dannysheridan** [comment]
image

**Comment by dannysheridan** [comment]
https://privategpt.docs.buildwithfern.com/

File: .github/workflows/fern/fern-check.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: ci
```
Comment (pabloogc) [inline .github/workflows/fern/fern-check.yml:1]:
```suggestion
name: fern-check
```

File: .github/workflows/fern/fern-check.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      - name: Install Fern
+        run:  npm install -g fern-api
+
+      - name: Check Fern API is valid
+        run: fern check
```
Comment (pabloogc) [inline .github/workflows/fern/fern-check.yml:19]: what does this check do? is it validation check or "fern is installed properly" check. If it's document validation I would move it to https://github.com/imartinez/privateGPT/blob/main/.github/workflows/tests.yml and add it as a mandatory condition

File: .github/workflows/fern/preview-docs.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+name: ci
```
Comment (pabloogc) [inline .github/workflows/fern/preview-docs.yml:1]:
```suggestion
name: deploy preview docs
```

File: .github/workflows/fern/publish-docs.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+name: ci
```
Comment (pabloogc) [inline .github/workflows/fern/publish-docs.yml:1]:
```suggestion
name: publish docs
```

File: .github/workflows/fern/publish-docs.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Setup node
+        uses: actions/setup-node@v3
+
+      - name: Download Fern
+        run: npm install -g fern-api
+
+      - name: Generate and Publish Docs
+        env:
+          FERN_TOKEN: ${{ secrets.FERN_TOKEN }}
+        run: fern generate --docs --log-level debug
```
Comment (pabloogc) [inline .github/workflows/fern/publish-docs.yml:24]: does this line upload too?

File: .github/workflows/fern/fern-check.yml (hunk 2 above)
Comment (dannysheridan) [reply .github/workflows/fern/fern-check.yml:19]: It validates the OpenAPI spec and the configuration files within `/fern`

Comment (dannysheridan) [reply .github/workflows/fern/fern-check.yml:19]: More details: https://docs.buildwithfern.com/overview/cli/fern-check

File: .github/workflows/fern/publish-docs.yml (hunk 5 above)
Comment (imartinez) [reply .github/workflows/fern/publish-docs.yml:24]: yes

**Comment by lopagela** [comment]
Done in https://github.com/imartinez/privateGPT/pull/1257

Thank you @dannysheridan !

**Pull request closed without merging**